subprojects/packagefiles/libopenjp2/meson_options.txt

```diff
@@ -12,3 +12,6 @@ option('build_java', type : 'boolean', value : false, description : 'Build the o
 option('build_testing', type : 'boolean', value : false, description : 'Build the tests.')
 option('build_pkgconfig_files', type : 'boolean', value : true, description : 'Build and install pkg-config files')
 option('with_astyle', type : 'boolean', value : false, description : 'Build with astyle')
+option('bin_png', type : 'boolean', value : false, description : 'Build binaries with libpng')
+option('bin_tiff', type : 'boolean', value : false, description : 'Build binaries with libtiff')
+option('bin_lcms2', type : 'boolean', value : false, description : 'Build binaries with liblcms2')
```

Comment:
Since these are used as `required: get_option('XXX')` for dependency lookups, setting `-Dbin_png=false` will actually build binaries with libpng, it just won't error out if it cannot.

It should be a feature option instead. The alternative is to check
```meson
if get_option('bin_png')
    png_dep = dependency('libpng')
else
    # always an empty not-found dependency
    png_dep = dependency('', required: false)
endif
```

Reply:
Thanks. This fixes the automagic dependencies, so I retract my change request.

I will caution that I think feature options are nicer and more modern / DRY, and I'd generally prefer to use them in cases like this.